sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2895,29 +2895,21 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
 
   def showCreateTableNotSupportedOnTempView(table: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1270",
+      errorClass = "UNSUPPORTED_SHOW_CREATE_TABLE.ON_TEMPORARY_VIEW",
```

Comment:
There is no `ON_TEMPORARY_VIEW` related test case because in `AstBuilder`, `allowTempView`parameter is set to `false` in the `visitShowCreateTable` method, and the code in `Analyzer`'s `ResolveRelations` rule `case u @ UnresolvedTableOrView(identifier, cmd, allowTempView)` will throw an `EXPECT_PERMANENT_VIEW_NOT_TEMP` exception.